src/main/scala/rise/elevate/rules/package.scala

```diff
@@ -40,7 +40,7 @@ package object rules {
 
   //TODO @rule
   def etaReduction()(implicit ev: Traversable[Rise]): Strategy[Rise] = {
-    case e@Lambda(x1, App(f, x2)) if x1 == x2 && !contains[Rise](x1).apply(f) => Success(f !: e.t)
+    case e@Lambda(x1, App(f, x2)) if x1 =~~= x2 && !contains[Rise](x1).apply(f) => Success(f !: e.t)
```

Comment:
I do not understand, why would comparing identifiers by name break this rule? Do we have different types for the same identifier name? Would this not be a bug?

Reply:
You are right! Using `==` didn't work for a while because `freeze` was broken, now that that is fixed `==` seems to work.